lib/ansible/modules/windows/win_share.ps1

```diff
@@ -149,32 +158,53 @@ Try {
 
         # need to (re-)create share
         If (-not $share) {
-            New-SmbShare -Name $name -Path $path
-            $share = Get-SmbShare $name -ErrorAction SilentlyContinue
+            if (-not $check_mode) {
+                New-SmbShare -Name $name -Path $path
```

Comment:
No worries, carry on  ! (But maybe document it briefly somewhere in the comments ?)